stmhal/mpconfigport.h

```diff
@@ -247,6 +196,73 @@ static inline mp_uint_t disable_irq(void) {
 // board specific definitions
 #include "mpconfigboard.h"
```

Comment:
I think we should move this include to the very start of this mpconfigport.h file (well not the very start, line 30, before any config option is set).  That makes sense because mpconfigboard.h is more specific than mpconfigport.h, just like mpconfigport.h is more specific than mpconfig.h (and mpconfig.h includes mpconfigport.h before anything else).

It also means that we can then start having default HW settings in mpconfigport.h which are only configured if not already done by mpconfigboard.h.